tests/tasks/teardown/awscli-vpc-delete.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: awscli-delete-vpc
+  namespace: tekton-pipelines
+spec:
+   description: |
+    This Task can be used to delete CloudFormation stack containing VPC resources that was used for EKS clusters.
+   params:
+    - name: stack-name
+      description: The name of the VPC name you want to delete.
+    - name: region
+      default: "us-west-2"
+   steps:
+      - name: awscli-delete-vpc
+        image: alpine/k8s:1.22.6
+        script: |
+          #!/bin/bash
+          echo "Approving KCM requests"
+          kubectl certificate approve $(kubectl get csr | grep "Pending" | awk '{print $1}')  2>/dev/null || true
+          aws sts get-caller-identity 
+          # Check if the stack exists
+          stack_exists=$(aws cloudformation --region $(params.region) describe-stacks --stack-name $(params.stack-name) 2>&1)
+          if [[ $stack_exists == *"does not exist"* ]]; then
+            echo "Error: Stack $(params.stack-name) does not exist."
+            exit 1
+          fi
```

Comment:
I hesitate to depend on the `error message` instead of `error code`. 
I would  rather reduce this whole condition to directly exit based on the `non-zero` exit code of the response for the `describe` call.

Reply:
@hakuna-matatah  If i understand correct you want me to display more like this instead of just `does not exist` response right ?

```
if [ $? -eq 0 ]; then
  printf "Stack exists\n"
  exit 0
else
  printf "Stack does not exist\n"
  exit 1

Reply:
+1
Also do we actually want to exit when the stack is not found. Exit with non-0 will cause pipeline to stop.

Reply:
What harish mean is to check the error code from `aws cloudformation describe-stacks`. Checking error message is a little fragile.

Reply:
>Also do we actually want to exit when the stack is not found. Exit with non-0 will cause pipeline to stop.

@mengqiy I think we should exit with non zero when stack not found as  it's very unexpected to have vpc stack disappear if every other task in test pipeline run was successful, we want to understand if those cases arise and not mask them. wdyt ?